Result/Result.swift

```diff
@@ -31,6 +31,15 @@ public enum Result<T, Error: ErrorType>: ResultType, CustomStringConvertible, Cu
 		}
 	}
 	
+	/// Constructs a result from a function that uses `throw`, failing with `Error` if throws, non-autoclosure version
+    	public init?(_ f: () throws -> T?) {
+        	do {
+            		guard let result = try f() else { return nil }
```

Comment:
Yes, I know. Even if there was no bag in Swift compiler, there is still a need of un-autoclosure version.